src/ui/public/timefilter/timefilter.js

```diff
@@ -20,142 +20,157 @@
 import _ from 'lodash';
 import moment from 'moment';
 import { calculateBounds, getTime } from './get_time';
-import '../state_management/global_state';
-import '../config';
-import { EventsProvider } from '../events';
+import { parseQueryString } from 'ui/timefilter/lib/parse_querystring';
 import { diffTimeFactory } from './lib/diff_time';
 import { diffIntervalFactory } from './lib/diff_interval';
+import { SimpleEmitter } from 'ui/utils/simple_emitter';
 import uiRoutes from '../routes';
-import { uiModules } from '../modules';
-import { createLegacyClass } from '../utils/legacy_class';
+import chrome from 'ui/chrome';
+
+class Timefilter extends SimpleEmitter {
+  constructor() {
+    super();
+    const self = this;
+    this.diffTime = diffTimeFactory(self);
```

Comment:
Can we please move these into the class definition for `TimeFilter`? I think that having the definition of this class spread out over multiple files makes it pretty tricky to tell what it's doing, like which events it emits and in what conditions.